planner/cascades/transformation_rules.go

```diff
@@ -14,15 +14,22 @@
 package cascades
 
 import (
+	"github.com/pingcap/tidb/expression"
 	plannercore "github.com/pingcap/tidb/planner/core"
 	"github.com/pingcap/tidb/planner/memo"
+	"github.com/pingcap/tidb/util/ranger"
 )
 
 // Transformation defines the interface for the transformation rules.
 type Transformation interface {
 	GetPattern() *memo.Pattern
-	Match(expr *memo.ExprIter) (matched bool)
-	OnTransform(old *memo.ExprIter) (new *memo.GroupExpr, eraseOld bool, err error)
+	Match(expr *memo.ExprIter) bool
```

Comment:
It seems that if a rule matches a node is determined by `Iter` according to the pattern of this rule, so this `Match` is unnecessary and can be removed?

Reply:
Please reference comment in https://github.com/pingcap/tidb/pull/11566/files#diff-46462632826dca717f2913bf43b93625R117 for explanation.